### PR TITLE
Rework Scene and GUI Editor connection

### PR DESCRIPTION
**Pull request by carolhmj**
Currently, when a GUI Editor instance is opened through the Inspector, all existing controls on the Scene are removed and then re-added to both the "live" scene and the editor scene. This has been causing variety of problems, such as:

- When a control in the editor is scaled, it doesn't update visually on the live scene
- If a control in the live scene is linked to a mesh, it loses that connection after the editor is opened
- Scroll viewers on the live scene lose scroll functionality once the editor is opened
- Controls on the live scene changed size because the editor scene could have a different size 

This PR introduces a new approach, such as when the editor is opened, the "live" scene's Controls are cloned and the clones added to the editor scene. Then, it periodically executes the opposite operation, cloning the editor controllers and adding them to the live scene.

### CODE REVIEW

**Comment by RaananW** [comment]
Snapshot available:

Snapshot stored with reference name:
refs/pull/13353/merge

Test environment:
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13353/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13353/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13353/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13353/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13353/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13353/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13353/merge#BCU1XR#0

File: packages/dev/gui/src/2D/controls/control.ts [hunk 1]
```diff
@@ -2323,6 +2323,27 @@ export class Control implements IAnimatable {
         this.getDescendants().forEach((child) => child._markAllAsDirty());
     }
 
+    public clone(host?: AdvancedDynamicTexture): Control {
```
Comment (sebavan) [inline packages/dev/gui/src/2D/controls/control.ts:2326]: missing ts doc

File: packages/dev/gui/src/2D/controls/control.ts [hunk 2]
```diff
@@ -2323,6 +2323,27 @@ export class Control implements IAnimatable {
...
+        this.serialize(serialization);
+
+        const controlType = Tools.Instantiate("BABYLON.GUI." + serialization.className);
+        const cloned = new controlType();
+        cloned.parse(serialization, host);
+
+        return cloned;
+    }
+
+    public parse(serializedObject: any, host?: AdvancedDynamicTexture): Control {
```
Comment (sebavan) [inline packages/dev/gui/src/2D/controls/control.ts:2337]: missing ts doc

**Review by sebavan — APPROVED** [review]
only nits :-)

File: packages/dev/gui/src/2D/controls/control.ts (hunk 1 above)
Comment (carolhmj) [reply packages/dev/gui/src/2D/controls/control.ts:2326]: Thanks for the reminder I ended up forgetting!

**Pull request merged**